bpf/include/bpf/verifier.h

```diff
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2016-2020 Authors of Cilium */
+
+#ifndef __BPF_VERIFIER__
+#define __BPF_VERIFIER__
+
+#include "compiler.h"
+
+/* relax_verifier is a dummy helper call to introduce a pruning checkpoint
+ * to help relax the verifier to avoid reaching complexity limits on older
+ * kernels.
+ */
+static __always_inline void relax_verifier(void)
+{
+#ifdef NEEDS_RELAX_VERIFIER
+       volatile int __maybe_unused id = get_smp_processor_id();
```

Comment:
Following up on last week discussion on the performance implications of using `ktime_get_ns()` I have switched to `get_smp_processor_id()` which should get compiled down to a single instruction.

To confirm that I had look at the generated asm.

test module:
```c
..
int init_module(void)
{
	volatile int x = smp_processor_id();
	return 0;
}
```

objdump output:
```
0000000000000000 <init_module>:
   0:	e8 00 00 00 00       	callq  5 <init_module+0x5>
			1: R_X86_64_PLT32	__fentry__-0x4
   5:	55                   	push   %rbp
   6:	48 89 e5             	mov    %rsp,%rbp
   9:	48 83 ec 08          	sub    $0x8,%rsp
   d:	65 8b 05 00 00 00 00 	mov    %gs:0x0(%rip),%eax        # 14 <init_module+0x14>
			10: R_X86_64_PC32	cpu_number-0x4
  14:	89 45 fc             	mov    %eax,-0x4(%rbp)
..
```

Reply:
/cc @borkmann